src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/ExternalMovementsSyncService.kt

```diff
@@ -163,6 +166,42 @@ class ExternalMovementsSyncService(
     } ?: run { telemetryClient.trackEvent("$TELEMETRY_PREFIX-outside-movement-deleted-ignored", telemetry) }
   }
 
+  suspend fun scheduledMovementInserted(event: ScheduledMovementEvent) = when (event.eventMovementType) {
+    "TAP" -> scheduledMovementInsertedTAP(event)
+    else -> log.info(("Ignoring scheduled movement event with type ${event.eventMovementType}"))
+  }
+
+  suspend fun scheduledMovementInsertedTAP(event: ScheduledMovementEvent) {
+    val (eventId, bookingId, prisonerNumber, eventMovementType) = event
+    val telemetry = mutableMapOf<String, Any>(
+      "offenderNo" to prisonerNumber,
+      "bookingId" to bookingId,
+      "nomisEventId" to eventId,
+      "eventMovementType" to eventMovementType,
+    )
+
+    if (event.doesOriginateInDps()) {
+      telemetryClient.trackEvent("$TELEMETRY_PREFIX-scheduled-movement-inserted-skipped", telemetry)
+      return
+    }
+
+    mappingApiService.getScheduledMovementMapping(eventId)
+      ?.also { telemetryClient.trackEvent("$TELEMETRY_PREFIX-scheduled-movement-inserted-ignored", telemetry) }
+      ?: run {
+        track("$TELEMETRY_PREFIX-scheduled-movement-inserted", telemetry) {
+          // TODO SDIT-3021 This should call getTemporaryAbsenceScheduledReturnMovement if the direction code is IN
+          nomisApiService.getTemporaryAbsenceScheduledMovement(prisonerNumber, eventId)
```

Comment:
Made a small mistake here - didn't ask for DIRECTION_CODE to be included in the XTAG event so don't know when to get the scheduled return movement.

This is important so we can tell DPS the relationship between scheduled movements, e.g. this return movement is a child of this outbound movement.